assets/queries/terraform/azure/aks_uses_azure_policies_addon_disabled/query.rego

```diff
@@ -26,37 +28,26 @@ CxPolicy[result] {
 }
 
 CxPolicy[result] {
-	cluster := input.document[i].resource.azurerm_kubernetes_cluster[name].addon_profile
-
-	not common_lib.valid_key(cluster, "azure_policy")
 
-	result := {
-		"documentId": input.document[i].id,
-		"resourceType": "azurerm_kubernetes_cluster",
-		"resourceName": tf_lib.get_resource_name(cluster, name),
-		"searchKey": sprintf("azurerm_kubernetes_cluster[%s].addon_profile", [name]),
-		"issueType": "MissingAttribute",
-		"keyExpectedValue": sprintf("'azurerm_kubernetes_cluster[%s].addon_profile.azure_policy' is defined and set to true", [name]),
-		"keyActualValue": sprintf("'azurerm_kubernetes_cluster[%s].addon_profile.azure_policy' is undefined or null", [name]),
-		"searchLine": common_lib.build_search_line(["resource", "azurerm_kubernetes_cluster", name, "addon_profile"], []),
-	}
-}
+	# after azurerm 3.0
 
-CxPolicy[result] {
 	cluster := input.document[i].resource.azurerm_kubernetes_cluster[name]
+	cluster.azure_policy_enabled == false
 
-	not common_lib.valid_key(cluster, "addon_profile")
 
 	result := {
 		"documentId": input.document[i].id,
 		"resourceType": "azurerm_kubernetes_cluster",
 		"resourceName": tf_lib.get_resource_name(cluster, name),
-		"searchKey": sprintf("azurerm_kubernetes_cluster[%s]", [name]),
-		"issueType": "MissingAttribute",
-		"keyExpectedValue": sprintf("'azurerm_kubernetes_cluster[%s].addon_profile' is defined and set to true", [name]),
-		"keyActualValue": sprintf("'azurerm_kubernetes_cluster[%s].addon_profile' is undefined or null", [name]),
-		"searchLine": common_lib.build_search_line(["resource", "azurerm_kubernetes_cluster", name], []),
-		"remediation": "addon_profile {\n\t\tazure_policy {\n\t\t\t enabled = true\n\t\t}\n\t}",
-		"remediationType": "addition",
+		"searchKey": sprintf("azurerm_kubernetes_cluster[%s].azure_policy_enabled", [name]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": sprintf("'azurerm_kubernetes_cluster[%s].azure_policy_enabled' is set to true", [name]),
+		"keyActualValue": sprintf("'azurerm_kubernetes_cluster[%s].azure_policy_enabled' is set to false", [name]),
+		"searchLine": common_lib.build_search_line(["resource", "azurerm_kubernetes_cluster", name, "azure_policy_enabled"], []),
+		"remediation": json.marshal({
+			"before": "false",
+			"after": "true"
+		}),
+		"remediationType": "replacement",
```

Comment:
Hello, @rndmh3ro!

Thank you so much for taking care of the azure 3.0 changes!

I think we should maintain the following "old" policy because this policy will only return a result if `addon_profile` is defined (which indicates that the user is using version before 3.0) but does not have `azure_policy` defined

```
# before azurerm 3.0
CxPolicy[result] {
	cluster := input.document[i].resource.azurerm_kubernetes_cluster[name].addon_profile

	not common_lib.valid_key(cluster, "azure_policy")

	result := {
		"documentId": input.document[i].id,
		"resourceType": "azurerm_kubernetes_cluster",
		"resourceName": tf_lib.get_resource_name(cluster, name),
		"searchKey": sprintf("azurerm_kubernetes_cluster[%s].addon_profile", [name]),
		"issueType": "MissingAttribute",
		"keyExpectedValue": sprintf("'azurerm_kubernetes_cluster[%s].addon_profile.azure_policy' is defined and set to true", [name]),
		"keyActualValue": sprintf("'azurerm_kubernetes_cluster[%s].addon_profile.azure_policy' is undefined or null", [name]),
		"searchLine": common_lib.build_search_line(["resource", "azurerm_kubernetes_cluster", name, "addon_profile"], []),
	}
}
```

Furthermore, I think we should add another policy that verifies that `addon_profile` and `azure_policy_enabled` are not defined (in order to verify that the AKS is not using Azure Policies). Since there is no default for `azure_policy_enabled` , I think this policy makes sense.

```
CxPolicy[result] {
	cluster := input.document[i].resource.azurerm_kubernetes_cluster[name]

	not common_lib.valid_key(cluster, "addon_profile") # before version 3.0
    not common_lib.valid_key(cluster, "azure_policy_enabled") # after version 3.0

	result := {
		"documentId": input.document[i].id,
		"resourceType": "azurerm_kubernetes_cluster",
		"resourceName": tf_lib.get_resource_name(cluster, name),
		"searchKey": sprintf("azurerm_kubernetes_cluster[%s]", [name]),
		"issueType": "MissingAttribute",
		"keyExpectedValue": sprintf("'azurerm_kubernetes_cluster[%s]' should use Azure Policies", [name]),
		"keyActualValue": sprintf("'azurerm_kubernetes_cluster[%s]' does not use Azure Policies", [name]),
		"searchLine": common_lib.build_search_line(["resource", "azurerm_kubernetes_cluster", name], []),
	}
}
```

Let me know what do you think 😊

Reply:
You're right! I'm still new to kics and rego so I appreciate your help.
I added the two tests. However I couldn't add another test - I guess just as the other PR it's not possible. If it is, please tell me how. :)

Reply:
I am glad to help! 😊  If you have any doubt or if you need anything, let me know!

You can add a `positive3.tf` and `positive4.tf` samples:

positive3.tf 
```
resource "azurerm_kubernetes_cluster" "positive3" {
  name                = "example-aks1"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  dns_prefix          = "exampleaks1"

  addon_profile {}
}
```


positive4.tf

```
resource "azurerm_kubernetes_cluster" "positive4" {
  name                = "example-aks1"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  dns_prefix          = "exampleaks1"
}
```

Reply:
Thanks! I tried that and it didn't work initially. But that was just because I forgot to also update the expected_result.json.

Not it's working!